### ISSUE TITLE
Technical, design reviews and copy-editing: Machine-aided detection of sources in Jinpingmei

### ISSUE THREAD

**Issue by inactinique** [C2DH/jdh-notebook#201]
Auhtor's repo: https://github.com/vierth/5QKJH5KQn9r3/tree/revisions (cloned from: https://github.com/jdh-observer/5QKJH5KQn9r3)

https://github.com/jdh-observer/5QKJH5KQn9r3

**Comment by eliselavy** [comment]
@biumiamy copy editing can be inserted

**Comment by AdamBela123** [comment]
![Image](https://github.com/user-attachments/assets/8a5831ea-af88-4ced-bac3-72b94b200c6a)

This correction does not change the text.
https://github.com/jdh-observer/5QKJH5KQn9r3/pull/7
Spelling mistakes in this pull request.

**Comment by biumiamy** [comment]
we found the problem, there are some more corrections that need to be done (but they are hidden in the Reviewing Pane

![Image](https://github.com/user-attachments/assets/479af788-4448-430c-8e4f-0bb314ba1c86)

> ![Image](https://github.com/user-attachments/assets/8a5831ea-af88-4ced-bac3-72b94b200c6a)
> 
> This correction does not change the text. [jdh-observer/5QKJH5KQn9r3#7](https://github.com/jdh-observer/5QKJH5KQn9r3/pull/7) Spelling mistakes in this pull request.

**Comment by biumiamy** [comment]
There are quite a few citations that dont seem very readable..

for example:
sklearn.feature_extraction.text.TfidfVectorizer. (n.d.). Scikit Learn. https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.TfidfVectorizer.html#sklearn.feature_extraction.text.TfidfVectorizer

<img width="805" height="609" alt="Image" src="https://github.com/user-attachments/assets/d7a5a6ca-c29e-4d9c-9f59-cc382a1e31b7" />

**Comment by biumiamy** [comment]
- design review okay
- license?

<img width="747" height="98" alt="Image" src="https://github.com/user-attachments/assets/8b733b42-92d7-4599-93c6-6ea0a11ab961" />

- need ask the author if the citations are correct this way.. 
- anyway, ready for final author check @inactinique

**Comment by inactinique** [comment]
Thanks!

- sent to author for final check with specif questions on bibliographic references,
- asked tweets,
- asked license to publish with license.

**Comment by eliselavy** [comment]
@biumiamy changes to be inserted

**Comment by biumiamy** [comment]
author email:

Abstract: The pseudonymous author of Jinpingmei spins a masterful and highly pornographic story out of a brief episode from an earlier novel the Water Margin and peppers the adapted narrative with intrusions of material from a dizzying array of external works with seemingly no regard for the boundaries of genre.

should be

The pseudonymous author of Jinpingmei spins a masterful and highly pornographic story out of a brief episode from an earlier novel Shuihu zhuan (Water Margin) and peppers the adapted narrative with intrusions of material from a dizzying array of external works with seemingly no regard for the boundaries of genre.

 

 

4: Roy should be Roy 1993.

 

As Patrick Hanan points out, this same lyric poem (with only very minor differences) can be found in the late fifteenth century Hong Pian’s 洪楩 Qingpingshan huaben 清平山堂話本, a collection of short stories referenced throughout Jinpingmei (Hanan 1963, 25).

should be

As Patrick Hanan points out, this same lyric poem (with only very minor differences) can be found in the late fifteenth century Hong Pian’s 洪楩 Qingpingshantang huaben (Huaben from Qingping Mountain Hall 清平山堂話本), a collection of short stories referenced throughout Jinpingmei (Hanan 1963, 25).

 

The cihua edition immediately pivots from this relatively short instance of intertextuality to a more widely recognized, and much longer, adaptation of material from the novel Water Margin (Shuihu zhuan 水滸傳).

should be

The cihua edition immediately pivots from this relatively short instance of intertextuality to a more widely recognized, and much longer, adaptation of material from the novel Shuihu zhuan (Water Margin 水滸傳).

 

 

6: Determinantive should be deterministic

 

12: One notable phenomenon is that quotes linked to unkown/contemporary are more common than those that predate Jinpingmei, but they also tend to be much shorter in length.

should be

One notable phenomenon is that quotes linked to unknown/contemporary works are more common than those that predate Jinpingmei, but they also tend to be much shorter in length.

 

 

14: Some phrases are rarer, appearing only a single time in Jinpingmei but appear widely across the corpus.

should be

Some phrases are rarer, appearing only a single time in Jinpingmei but widely elsewhere within the corpus.

 

Detecting which of the multiple texts in which a quote appears is the actual source can be difficult.

should be

It can be difficult to detect the actual source among the multiple texts a quote might appear in.

 

20: References that can only be sourced to unknown materials are abundant but more intermittent than those from earlier texts and computational analysis represents fascinating possibilities for understanding their likely origins.

should be

References that can only be sourced to contemporary/unknown materials are abundant but more intermittent than those from earlier texts and computational analysis represents fascinating possibilities for understanding their likely origins.

 

22: Jinpingmei should be italicized

 

Finally, I have the material that was roughly contemporary with the novel, which form the body of materials that necessitate this kind of investigation (because available chronological evidence is not enough to ascertain the potential textual relationship).

should be

Finally, I have the material that was roughly contemporary with the novel, which form the body of materials that necessitate this kind of investigation because available chronological evidence is not enough to ascertain the potential textual relationship.

 

 

24: As such, I turn to “bag-of-words” representations of texts (technically bag of n-grams) (Underwood 2013).

should be

As such, I turn to “bag-of-words” representations of texts, or more precisely "bag of n-consecutive characters in a row (also known as n-grams)," where n is normally one (Underwood 2013).

 

"simple n-gram frequencies" n should be italicized

 

26 (This whole paragraph has a few small issues): The approach I take to predicting the likely source of a quote (which I will refer to as a substring here, as a smaller section of the entire string the represents the whole document) rests on the intuition that the quote substring that represents a given instance of intertextuality is going to be more stylistically similar to the text it originally came from than the text that appropriates it. As such, I randomly extract an equal number of substrings from each document while ensuring that I do not grab text that is shared between the two documents to train my models (this is to say that I ensure that I do not use the shared quotes, which I want the model to predict the origins of, when training the models). Because the quotes I am comparing are generally between 10 and 250 or so characters long, I ensure that the substrings I randomly extract fall at some random length between these two extremes (though the results do contain notably longer quotations as well).

 

should be

 

The approach I take to predicting the likely source of a quote, which I will refer to as a "substring” here—a subsection of the string that is the text of the whole document—rests on the intuition that the quote substring that represents a given instance of intertextuality is going to be more stylistically similar to the text it originally came from than the text that appropriates it. As such, to train my models I randomly extract an equal number of substrings from each document while ensuring that I do not include any intertextual material. I do this to ensure that I do not train the models on the shared quotes that are the subject of my investigation. Because the quotes I am comparing are generally between 10 and 250 or so characters long, I ensure that the substrings I randomly extract fall at some random length between these two extremes. Of course, the actual intertextual results do contain some quotations that are notably longer than this, but limiting training to 250-character segments does not appear to affect the accuracy of the algorithm.

 

37: Yujing xintan 玉鏡新譚 should be Yujing xintan (A New Discussion of the Jade Mirror 玉鏡新譚)

 

Jingshi yinyangmeng 警示陰陽夢

should be

Jingshi yinyang meng (A Dream of Light and Dark to Admonish the World 警示陰陽夢)

 

 

Vierthaler n.d. should be Vierthaler 2016

 

41: There is some overlap in the central part of this figure when sections from each text appear in similar spaces, but it is still distinct.

Should be

There is some overlap in the central part of this figure where sections from each text appear in similar spaces, but overall they are distinct.

 

 

47 (the closing parenthesis in this sentence should be deleted): We would expect the author of the Jingshi yinyang meng to edit some quotations they take from the Yujing xintan to better fit the style of the Jingshi).

 

 

49: This suggests that their distributions of frequent terms are more similar to each other than what is seen in the of the Wei texts.

Should be

This suggests that the distributions of frequent terms found in these novels are more similar to each other than what is seen in the Wei texts.

 

53: Again, significant variance are hidden from view in these types of visualizations.

should be

Again, significant variance is hidden from view in these types of visualizations.

 

55: The classifier struggles to stay much around 53 percent accuracy, and the edited sections of text found within Jinpingmei are associated with Shuihu zhuan just over 50 percent of the time.

should be

The classifier struggles to stay much above 53 percent accuracy, and the edited sections of text found within Jinpingmei are associated with Shuihu zhuan just over 50 percent of the time.

 

58 remove parentheses in this sentence: First, we can expect some noise simply because the exact version of the Water Margin the author likely used to compose Jinpingmei is not present in the corpus (and does not seem to exist anymore).

 

58: But Jinpingmei’s author copies these chapters nearly verbatim but alters the story such that Ximen Qing and Pan Jinlian initially escape Wu’s wrath.

Should be

Jinpingmei’s author copies these chapters nearly verbatim but alters the story such that Ximen Qing and Pan Jinlian initially escape Wu’s wrath.

 

64: Another intriguing example of intertextuality exists between Jinpingmei and Li Kaixian’s work Cinüe 詞謔, a rather obscure piece of dramatic criticism.

should be

Another intriguing example of intertextuality exists between Jinpingmei and Li Kaixian’s work Cinüe (Jesting on Lyrics 詞謔), a piece of dramatic criticism.

 

72: While this is disappointing in terms of the potential authorship of Jinpingmei, as we cannot use this as evidence that Li had a copy of the novel, it does comport well with the expectations we would have upon a careful expectation of the Cinüe as a piece of dramatic criticism and upon considering the deeper layers of quotation that appear in these sections of Jinpingmei.

should be

This is disappointing in terms of the potential authorship of Jinpingmei, because we cannot use this as evidence that Li had a copy of the novel. However, it comports well with the expectations we should have after considering that the Cinue is a piece of dramatic criticism and upon observing that the intertextuality algorithm reveals even deeper layers of quotation occurring in these sections.

 

74: This means that I train an independent model for each comparison (so Jinpingmei vs. Water Margin, Jinpingmei vs. Cinüe, Jinpingmei vs. Qingpingshan huaben and so on).

should be

This means that I train an independent model for each comparison (so Jinpingmei vs. Shuihu zhuan, Jinpingmei vs. Cinüe, Jinpingmei vs. Qingpingshantang huaben and so on).

 

 

The main utility of this approach is ease of interpretability and accuracy as multi-class models become much more difficult to work with as the number of classes increase. 

Should be two sentences:

The main utility of this approach is ease of interpretability and accuracy. Multi-class models become much more difficult to work with as the number of classes increase.

 

I should note that training so many models is computationally intense, but the method remains relatively scalable as each pair-wise classifier takes somewhere between .005 seconds and 2 seconds to train in this notebook run on a 2024 MacBook Pro (I could significantly optimize if I need to perform larger-scale analysis).

Should also be multiple sentences:

I should note that training so many models is somewhat computationally intense, but the method remains relatively scalable. Each pair-wise classifier takes somewhere between .005 seconds and 2 seconds to train inside this notebook environment on a 2024 MacBook Pro. There is significant room for optimization if I need to perform larger-scale analysis.

 

78: An alternative explanation may be that in half the cases an author edited the shared quotes enough for the model to judge them as more similar to the text doing the quoting.

should be

An alternative explanation may be that in half the cases an author edited the shared quotes enough for the model to judge them as more similar to the quoting text than as their real source.

 

Among the missed cases that satisfy both of these criteria is a poem found in a commentary on a Daoist text called the Taishang laojun shuochang qingjing jingzhu 太上老君說常清靜經注

should be (this includes a fixed Chinese character)

Among the missed cases include a poem found in a commentary on a Daoist text called the Taishang laojun shuochang qingjing jingzhu (Commentary on the Classic of Everlasting Purity and Tranquility as told by Supreme Lord Lao 太上老君說常清靜經註).

 

In this case, the tone and content of the poem line up clearly with Jinpingmei and less so with the broader context of Taishang laojun.

add a sentence after this:

Additionally, this poem is found in twenty-eight works in the corpus, including Shuihu zhuan. It can ultimately be traced to the Tang and is found in the Quantangshi (Complete Poems of the Tang Dynasty 全唐詩) under works by Lü Dongbin 呂洞賓 (fl. 9th century).

 

 

Technically, this is the average distance the shared quotes are from the hyperplane that separates Jinpingmei from the other text.

should be

Technically, this is the average of how far all shared quotes are from the boundary the model uses to separate the two classes.

 

The quote from the Qingpingshantang huaben that appears at the beginning of this article (and the novel itself) is a good example: the intertextuality algorithm identifies three texts in which the opening poem appears: the Qingpingshantang huaben, the Quansongci 全宋詞, and the Shuofu 說郛.

should be

The quote from the Qingpingshantang huaben that appears at the beginning of this article is a good example. The intertextuality algorithm identifies three texts in which the opening poem appears: the Qingpingshantang huaben, the Quansongci (Complete Lyric Poems of the Song Dynasty 全宋詞), and the Shuofu (Purlieus of Exposition 說郛, a translation by Victor Mair via Christopher Atwood, (Atwood, 2017)).

 

Full citation of [https://sino-platonic.org/complete/spp271_tao_zongyi_shuofu.pdf](https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fsino-platonic.org%2Fcomplete%2Fspp271_tao_zongyi_shuofu.pdf&data=05%7C02%7Cmirjam.pfeiffer%40uni.lu%7C3d61f54337f64d437df708ddeb9368b1%7C445a9c950f9d49539db1bc4a45dd1220%7C0%7C0%7C638925740575106127%7CUnknown%7CTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7C0%7C%7C%7C&sdata=E7vq5z2kare0YyLICWIOkra9aC60uepCvdvxihfDlnc%3D&reserved=0):

Atwood, C. (2017). The Textual History of Tao Zongyi's Shuofu: Preliminary Results of Stemmatic Research on the Shengwu qingzheng lu. Sino-Platonic Papers, 271, 1-73.

 

84: The results are fractured because the people featured in the stories are different and causing the intertextuality algorithm to return multiple shorter substrings.

should be

The results are fractured because the people featured in the stories are different, causing the intertextuality algorithm to return multiple shorter substrings.

 

Zhongguo Su Wen Ku should be Zhongguo suwen ku

Full citation: Liu, J., & Airusheng shuzihua jishu yanjiu zhongxin. (2011). Zhongguo suwen ku. Beijing airusheng shuzihua jishu yanjiu zhongxin.

 

 

Zhongguo Tongsu Xiaoshuo Zongmu

should be

Zhongguo tongsu xiaoshuo zongmu tiyao

Full citation:

Zhongguo tongsu xiaoshuo zongmu tiyao. (1990). Zhongguo wenlian chuban gongsi.

 

86: Jinpingmei should be italicized (2x)

 

General: please change all cases of "Water Margin" to "Shuihu zhuan” (except for the very first one). I want this to happen to bring it in stylistic accord with all the other titles.

 All instances of Jingshi yinyangmeng should be Jingshi yinyang meng

 

Please replace “layeredintertextuality.png” with the updated version I’ve attached here. This should be figure 11, found in the media folder

**Comment by inactinique** [comment]
> [@inactinique](https://github.com/inactinique) is sending another reminder to the author

Author answered and sent a pull request

**Comment by biumiamy** [comment]
I think its ready for publishing @eliselavy

**Comment by eliselavy** [comment]
> sent

copyright save on atlas
article updated with good copyright [CC-BY-NC-ND](https://creativecommons.org/licenses/by-nc-nd/4.0/)
wait for tweet
